### ISSUE TITLE
nixosTests.containers-{ip,imperative} are failing on master

### ISSUE THREAD

**Issue by mweinelt** [NixOS/nixpkgs#167001]
Both `nixosTests.containers-ip` and `nixosTests.containers-imperative` are failing on master, which prevents nixos-unstable from advancing.

Hydra log:
https://hydra.nixos.org/build/171257813/nixlog/1

Bisected to:
```
a3e0698bf6fd676d96b89bdb4cd54c73ea502746 is the first bad commit
commit a3e0698bf6fd676d96b89bdb4cd54c73ea502746
Author: Janne Heß <janne@hess.ooo>
Date:   Wed Mar 23 15:56:23 2022 +0100

    nixos/systemd: Switch to ManagerEnvironment=
    
    This accomplishes multiple things:
    - Allows us to start systemd without stage-2-init.sh. This was not
      possible before because the environment would have been wrong
    - `systemctl daemon-reexec` also changes the environment, giving us
      newer tools for the fs packages
    - Starts systemd in a fully clean environment, making everything more
      consistent and pure

 nixos/modules/system/boot/stage-2-init.sh |  8 ++------
 nixos/modules/system/boot/stage-2.nix     | 11 ++---------
 nixos/modules/system/boot/systemd.nix     | 23 +++++++++++++++++++++++
 nixos/tests/systemd.nix                   |  4 ++++
 4 files changed, 31 insertions(+), 15 deletions(-)
```

Introduced in #165453 

@ajs124 @flokli @dasJ

**Comment by vcunat** [comment]
Revert for now in order to get the zlib update to `unstable-small`?  (also needs merging `staging-next` and maybe fixing additional stuff)

**Comment by dasJ** [comment]
Fix: #167019
